Sources/TuistKit/Mappers/Factories/ProjectMapperFactory.swift

```diff
@@ -79,6 +79,8 @@ public final class ProjectMapperFactory: ProjectMapperFactorying {
 
         // Signing
         mappers.append(SigningMapper())
+        
+        mappers.append(TargetProjectMapper(mapper: ExplicitDependencyTargetMapper()))
```

Comment:
Will this apply to all generated projects by tuist by default or in some cases like caching only? 🤔 

I'd be very cautious about introducing an additional script phase to all projects. Is it trying to solve a problem every project has or a specific edge case? would be helpful to understand the issue better.

Reply:
This will be hidden behind a feature flag or a config option, initially. And based on the feedback we receive, we can think of rolling it out further or adding this mapper by default for some commands like `tuist cache warm`.